consensus/reactor.go

```diff
@@ -338,7 +338,15 @@ func (conR *Reactor) ReceiveEnvelope(e p2p.Envelope) {
 		case *BlockPartMessage:
 			ps.SetHasProposalBlockPart(msg.Height, msg.Round, int(msg.Part.Index))
 			conR.Metrics.BlockParts.With("peer_id", string(e.Src.ID())).Add(1)
-			schema.WriteBlockPart(conR.traceClient, msg.Height, msg.Round, e.Src.ID(), msg.Part.Index, schema.TransferTypeDownload)
+			part, err := msg.Part.ToProto() // this conversion is only needed to get the part size
+			// consistent with how it is measured in the gossipDataRoutine
+			partSize := -1
+			if err != nil {
+				partSize = part.Size()
+			}
```

Comment:
[not blocking] Does this intentionally avoid checking for an error?

Reply:
Yes, it is intentional, because the earlier conversion (.toProto()) is merely for tracing data in influx DB, which is not a critical part of the consensus operation. Nevertheless, it is not expected to get any error.

Reply:
> Yes, it is intentional, because the earlier conversion (.toProto()) is merely for measuring the size and tracing it in influx DB, which is not a critical part of the consensus operation.

Makes sense! Since it is not critical to the consensus operation, I think this could not propagate the error and log it instead but this feels kinda hacky:

```go
part, err := msg.Part.ToProto()
if err != nil {
    // log the error
} else {
    schema.WriteBlockPart(conR.traceClient, msg.Height, msg.Round, e.Src.ID(), msg.Part.Index, schema.TransferTypeDownload, int64(part.Size()))
}
```
I'm not familiar with this code so defer to what you already have.